subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_df.c

```diff
@@ -234,58 +234,55 @@ static inline void radio_df_ctrl_set(uint8_t cte_len,
 				     uint8_t switch_spacing,
 				     uint8_t sample_spacing)
 {
-	uint32_t conf;
-
 	/* Complete setup is done on purpose, to be sure that there isn't left
 	 * any unexpected state in the register.
 	 */
-	conf = ((((uint32_t)cte_len << RADIO_DFECTRL1_NUMBEROF8US_Pos) &
-				       RADIO_DFECTRL1_NUMBEROF8US_Msk) |
-		((uint32_t)RADIO_DFECTRL1_DFEINEXTENSION_CRC <<
-			RADIO_DFECTRL1_DFEINEXTENSION_Pos) |
-		((uint32_t)switch_spacing << RADIO_DFECTRL1_TSWITCHSPACING_Pos) |
-		((uint32_t)NRF_RADIO_DFECTRL_SAMPLE_SPACING_1US <<
-			RADIO_DFECTRL1_TSAMPLESPACINGREF_Pos) |
-		((uint32_t)NRF_RADIO_DFECTRL_SAMPLE_TYPE_IQ <<
-			RADIO_DFECTRL1_SAMPLETYPE_Pos) |
-		((uint32_t)sample_spacing << RADIO_DFECTRL1_TSAMPLESPACING_Pos) |
-		(((uint32_t)0 << RADIO_DFECTRL1_AGCBACKOFFGAIN_Pos) &
-				 RADIO_DFECTRL1_AGCBACKOFFGAIN_Msk));
-
-	NRF_RADIO->DFECTRL1 = conf;
+	const nrf_radio_dfectrl_conf_t dfe_conf = {
+		.dfe_len = cte_len,
+		.extension = NRF_RADIO_DFECTRL_EXTENSION_CRC,
+		.switch_spacing = switch_spacing,
+		.spacing_ref = NRF_RADIO_DFECTRL_SAMPLE_SPACING_1US,
+		.sample_type = NRF_RADIO_DFECTRL_SAMPLE_TYPE_IQ,
+		.sample_spacing = sample_spacing,
+		.gain_steps = 0,
+		.switch_offset = 0,
+		.sample_offset = 0
+	};
+
+	nrf_radio_dfectrl_configure(NRF_RADIO, &dfe_conf);
```

Comment:
Until `DFECTRL2` is not updated then this code adds unnecessary instructions that may be saved. This function is executed in interrupt context. Policy for Nordic LLL is to save instructions in such situations.
Please change it back to previous version.